synapse/rest/client/room.py

```diff
@@ -757,6 +760,14 @@ async def on_POST(
 
         await self.room_member_handler.forget(user=requester.user, room_id=room_id)
 
+        forgotten = await self.store.is_locally_forgotten_room(room_id)
+        if forgotten:
+            run_as_background_process(
+                "purge_room_after_last_forget",
+                self.storage.purge_events.purge_room,
+                room_id,
+            )
```

Comment:
This is an OK first step, but it does mean that if the server gets restarted before the purge is completed it won't be retried.

One option would be to check if we need to purge the room and if so write that to a DB table, and once purged we can remove it from the table.

That can be a separate PR though I think